examples/storage-disk-advanced/src/main.rs

```diff
@@ -0,0 +1,189 @@
+    ctx: SimulationContext,
+    start_time: f64,
+}
+
+#[derive(Clone, Serialize)]
+struct Start {}
+
+#[derive(Clone, Serialize)]
+struct Ticker {}
+
+impl User {
+    fn new(disk: Rc<RefCell<Disk>>, ctx: SimulationContext) -> Self {
+        Self {
+            disk,
+            requests: HashMap::new(),
+            ctx,
+            start_time: 0.,
+        }
+    }
+}
+
+impl EventHandler for User {
+    fn on(&mut self, event: Event) {
+        cast!(match event.data {
+            Start {} => {
+                self.start_time = self.ctx.time();
+                for i in 1..READ_ITERATIONS {
+                    self.requests.insert(self.disk.borrow_mut().read(10, self.ctx.id()), i);
+                }
+                for i in 1..WRITE_ITERATIONS {
+                    self.requests
+                        .insert(self.disk.borrow_mut().write(10, self.ctx.id()), READ_ITERATIONS + i);
+                }
+            }
+            DataReadCompleted { request_id, size: _ } => {
+                log_info!(
+                    self.ctx,
+                    "Read iteration #{} completed. Elapsed time = {}",
+                    self.requests[&request_id],
+                    self.ctx.time() - self.start_time,
+                );
+            }
+            DataReadFailed { request_id, error } => {
+                log_info!(
+                    self.ctx,
+                    "Read iteration #{} failed. Error: {}",
+                    self.requests[&request_id],
+                    error
+                );
+            }
+            DataWriteCompleted { request_id, size: _ } => {
+                log_info!(
+                    self.ctx,
+                    "Write iteration #{} completed. Elapsed time = {}",
+                    self.requests[&request_id] - READ_ITERATIONS,
+                    self.ctx.time() - self.start_time,
+                );
+            }
+            DataWriteFailed { request_id, error } => {
+                log_info!(
+                    self.ctx,
+                    "Write iteration #{} failed. Error: {}",
+                    self.requests[&request_id],
+                    error
+                );
+            }
+        })
+    }
+}
+
+fn main() {
+    Builder::from_default_env()
+        .format(|buf, record| writeln!(buf, "{}", record.args()))
+        .init();
+
+    let mut sim = Simulation::new(SEED);
+    let root = sim.create_context("root");
+
+    let mut simple_spec = DiskSpec::default();
+    simple_spec
+        .set_capacity(DISK_CAPACITY)
+        .set_constant_read_bw(DISK_READ_BW)
+        .set_constant_write_bw(DISK_WRITE_BW);
+
+    let simple_disk = rc!(refcell!(Disk::new(simple_spec, sim.create_context(SIMPLE_DISK_NAME))));
+    sim.add_handler(SIMPLE_DISK_NAME, simple_disk.clone());
+
+    println!("Starting simple user...");
+
+    let simple_user = rc!(refcell!(User::new(simple_disk, sim.create_context(SIMPLE_USER_NAME))));
```

Comment:
Посколько код User один и тот в обоих запусках, отличия только в самом диске, то я бы убрал отдельные названия для юзера/клиента:

```suggestion
    let client = rc!(refcell!(User::new(simple_disk, sim.create_context(CLIENT_NAME))));
```

Reply:
А как их в логах отличать тогда?